pyccl/pk2d.py

```diff
@@ -104,6 +108,62 @@ def __init__(self, pkfunc=None, a_arr=None, lk_arr=None, pk_arr=None,
         check(status)
         self.has_psp = True
 
+    @classmethod
+    def pk_from_model(Pk2D, cosmo, model):
+        """`Pk2D` constructor returning the power spectrum associated with
+        a given numerical model.
+
+        Args:
+            cosmo (:class:`~pyccl.core.Cosmology`): A Cosmology object.
+            model (:obj:`str`): model to use. Three models allowed:
+                `'bbks'` (Bardeen et al. ApJ 304 (1986) 15).
+                `'eisenstein_hu'` (Eisenstein & Hu astro-ph/9710252).
+                `'emu'` (arXiv:1508.02654).
+        """
+        pk2d = Pk2D(empty=True)
+        status = 0
+        if model == 'bbks':
+            cosmo.compute_growth()
+            ret = lib.compute_linpower_bbks(cosmo.cosmo, status)
+        elif model == 'eisenstein_hu':
+            cosmo.compute_growth()
+            ret = lib.compute_linpower_eh(cosmo.cosmo, status)
+        elif model == 'emu':
+            ret = lib.compute_power_emu(cosmo.cosmo, status)
+        else:
+            raise ValueError("Unknown model %s " % model)
+
+        if np.ndim(ret) == 0:
+            status = ret
+        else:
+            pk2d.psp, status = ret
+
+        check(status, cosmo)
+        pk2d.has_psp = True
+        return pk2d
+
+    @classmethod
+    def halofit_it(Pk2D, cosmo, pk_linear):
```

Comment:
This function name is clever but a little hard to parse. Maybe `apply_halofit` or `compute_halofit`?

Reply:
Bummer :-P
Yep, changed to `apply_halofit`